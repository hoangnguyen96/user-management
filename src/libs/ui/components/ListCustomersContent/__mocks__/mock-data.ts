import { USER_ROLE } from "../../../../constants";

export const LIST_CONTENT = [
  {
    fullName: "Jane Cooper",
    role: USER_ROLE.ADMIN,
    company: "Microsoft",
    phoneNumber: "(225) 555-0118",
    email: "jane@microsoft.com",
    country: "United States",
    status: true,
    username: "admin",
    password: "123456",
    id: "1",
  },
  {
    fullName: "Floyd Miles",
    role: USER_ROLE.CONSUMER,
    company: "Yahoo",
    phoneNumber: "(205) 555-0100",
    email: "floyd@yahoo.com",
    country: "Kiribati",
    status: false,
    username: "consumer1",
    password: "123456",
    id: "2",
  },
  {
    fullName: "Ronald Richards",
    role: USER_ROLE.CONSUMER,
    company: "Adobe",
    phoneNumber: "(302) 555-0107",
    email: "ronald@adobe.com",
    country: "Israel",
    status: false,
    username: "consumer2",
    password: "123456",
    id: "3",
  },
  {
    fullName: "Marvin McKinney",
    role: USER_ROLE.CONSUMER,
    company: "Tesla",
    phoneNumber: "(252) 555-0126",
    email: "marvin@tesla.com",
    country: "Iran",
    status: true,
    username: "consumer3",
    password: "123456",
    id: "4",
  },
  {
    fullName: "Jerome Bell",
    role: USER_ROLE.CONSUMER,
    company: "Google",
    phoneNumber: "(629) 555-0129",
    email: "jerome@google.com",
    country: "Réunion",
    status: true,
    username: "consumer4",
    password: "123456",
    id: "5",
  },
  {
    fullName: "Kathryn Murphy",
    role: USER_ROLE.CONSUMER,
    company: "Micro",
    phoneNumber: "(406) 555-0120",
    email: "kathryn@microsoft.com",
    country: "Réunion",
    status: true,
    username: "consumer5",
    password: "123456",
    id: "6",
  },
  {
    fullName: "Jacob Jones",
    role: USER_ROLE.CONSUMER,
    company: "Amazon",
    phoneNumber: "(208) 555-0112",
    email: "jacob@yahoo.com",
    country: "Brazil",
    status: true,
    username: "consumer6",
    password: "123456",
    id: "7",
  },
  {
    fullName: "Kristin Watson",
    role: USER_ROLE.CONSUMER,
    company: "Facebook",
    phoneNumber: "(704) 555-0127",
    email: "kristin@facebook.com",
    country: "Åland Islands",
    status: false,
    username: "consumer7",
    password: "123456",
    id: "8",
  },
  {
    fullName: "Kris Leo",
    role: USER_ROLE.CONSUMER,
    company: "Facebook",
    phoneNumber: "(704) 555-0112",
    email: "krisleo@facebook.com",
    country: "Sahara",
    status: false,
    username: "consumer7",
    password: "123456",
    id: "9",
  },
];
