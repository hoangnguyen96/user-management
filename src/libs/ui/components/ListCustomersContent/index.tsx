import {
  ChangeEvent,
  memo,
  useCallback,
  useEffect,
  useMemo,
  useState,
  useTransition,
} from "react";
import { Box } from "@mui/material";

// Constants
import { SORT_DATA_CUSTOMERS } from "@app/constants";

// Apis
import { useGetCustomerById, useGetListCustomers } from "@app/api";

// Models
import { UserResponse } from "@app/models";

// Utils
import {
  dividePaginationCustomers,
  filterListCustomersByName,
  sortCustomersList,
} from "@app/utils";

// Components
import ListItemCustomers from "./ListItemCustomers";
import HeadingSearch from "../HeadingSearch";

interface ListCustomersContentProps {
  isAdmin?: boolean;
  id: string;
}

const ListCustomersContent = ({
  isAdmin = false,
  id,
}: ListCustomersContentProps) => {
  const [pagination, setPagination] = useState<number>(0);
  const [listPage, setListPage] = useState<UserResponse[][]>([]);
  const [textSearch, setTextSearch] = useState<string>("");
  const [filterText, setFilterText] = useState<string>("");
  const [isPending, startTransition] = useTransition();
  const [selectionValue, setSelectionValue] = useState(
    SORT_DATA_CUSTOMERS[0].value
  );

  const {
    isLoading: isLoadingListAll,
    refetch: refetchList,
    data: customersListData,
    errorMessage: errorGetList,
  } = useGetListCustomers();

  const { isLoading: isLoadingListById, data: customersListDataById } =
    useGetCustomerById(id);

  const customersList = useMemo(() => {
    if (!isAdmin) {
      return [customersListDataById] || [];
    }

    return errorGetList || customersListData.length <= 0
      ? []
      : customersListData;
  }, [isAdmin, customersListData, customersListDataById]);

  // Search Bar
  useEffect(() => {
    if (filterText) {
      const listByName = filterListCustomersByName(customersList, filterText);
      const list = dividePaginationCustomers(listByName);

      return setListPage(list);
    }

    const listAll = dividePaginationCustomers(customersList);
    return setListPage(listAll);
  }, [filterText]);

  // Selection Bar
  useEffect(() => {
    if (selectionValue !== "default") {
      const sortedCustomersList = sortCustomersList(
        customersList,
        selectionValue
      );

      const list = dividePaginationCustomers(sortedCustomersList);
      return setListPage(list);
    }

    const listAll = dividePaginationCustomers(customersList);
    return setListPage(listAll);
  }, [selectionValue]);

  // Reset List all
  useEffect(() => {
    const listAll = dividePaginationCustomers(customersList);
    setListPage(listAll);
  }, [isLoadingListAll, isLoadingListById, customersList]);

  const handleChangePagination = useCallback(
    (_event: ChangeEvent<unknown>, value: number) => {
      setPagination(value - 1);
    },
    [setPagination]
  );

  const handleOnChangeText = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      setTextSearch(e.target.value);

      startTransition(() => {
        setFilterText(e.target.value);
      });
    },
    [setFilterText, startTransition]
  );

  const currentPageData = listPage[pagination] || [];

  return (
    <Box
      sx={{
        borderRadius: "24px",
        backgroundColor: "white",
        boxShadow: "1px 1px 2px #cccccc",
      }}
    >
      <HeadingSearch
        title="All Customers"
        subTitle="Active Members"
        textSearch={textSearch}
        list={SORT_DATA_CUSTOMERS}
        handleOnChangeText={handleOnChangeText}
        setSelectionValue={setSelectionValue}
        setTextSearch={setTextSearch}
      />

      {/* List */}
      <ListItemCustomers
        isAdmin={isAdmin}
        isLoading={isLoadingListAll}
        isPending={isPending}
        listCurrent={currentPageData}
        listPage={listPage}
        error={errorGetList}
        pagination={pagination}
        paginationList={customersList}
        refetchList={refetchList}
        onChangePagination={handleChangePagination}
      />
    </Box>
  );
};

export default memo(ListCustomersContent);
