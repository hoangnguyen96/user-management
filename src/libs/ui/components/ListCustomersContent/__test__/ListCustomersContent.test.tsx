import { renderWithQueryClient } from "@app/ui/test-utils";

// Models
import { UserResponse } from "@app/models";

// Component
import ListCustomersContent from "..";
import { USER_ROLE } from "@app/constants";
import { fireEvent, waitFor } from "@testing-library/react";
import { LIST_CONTENT } from "../__mocks__/mock-data";

const mockCustomer: UserResponse = {
  fullName: "Jane Cooper",
  role: USER_ROLE.ADMIN,
  company: "Microsoft",
  phoneNumber: "(225) 555-0118",
  email: "jane@microsoft.com",
  country: "United States",
  status: true,
  username: "admin",
  password: "123456",
  id: "1",
};

jest.mock("@app/api", () => ({
  useGetCustomerById: jest.fn(() => ({
    isLoading: false,
    refetch: jest.fn(),
    data: mockCustomer,
  })),
  useGetListCustomers: jest.fn(() => ({
    isLoading: false,
    refetch: jest.fn(),
    data: LIST_CONTENT,
    errorMessage: "",
  })),
  useUpdateCustomer: jest.fn(() => ({
    isPending: false,
    mutate: jest.fn(),
  })),
  useDeleteCustomer: jest.fn(() => ({
    isPending: false,
    mutate: jest.fn(),
  })),
}));

describe("ListCustomersContent component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render snapshot correctly", () => {
    expect(
      renderWithQueryClient(<ListCustomersContent isAdmin={true} id="1" />)
    ).toMatchSnapshot();
  });

  it("Should handle text input change correctly", async () => {
    const { getByPlaceholderText } = renderWithQueryClient(
      <ListCustomersContent isAdmin={true} id="1" />
    );

    const input = getByPlaceholderText("Searchâ€¦") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "John" } });

    await waitFor(() => {
      expect(input.value).toBe("John");
    });
  });
});
