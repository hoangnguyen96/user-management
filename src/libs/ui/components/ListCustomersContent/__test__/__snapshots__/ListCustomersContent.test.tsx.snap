// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListCustomersContent component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1dachdb"
      >
        <div
          class="MuiStack-root css-133lg72-MuiStack-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-15gon46-MuiTypography-root"
            >
              All Customers
            </h4>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1ge8cc0-MuiTypography-root"
            >
              Active Members
            </h6>
          </div>
          <div
            class="MuiStack-root css-qoybat-MuiStack-root"
          >
            <div
              class="MuiBox-root css-79elbk"
            >
              <div
                class="MuiStack-root css-1ibunwk-MuiStack-root"
              >
                <svg
                  fill="none"
                  height="24px"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 19C15.4183 19 19 15.4183 19 11C19 6.58172 15.4183 3 11 3C6.58172 3 3 6.58172 3 11C3 15.4183 6.58172 19 11 19Z"
                    stroke="#7e7e7e"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                  <path
                    d="M21 21L16.65 16.65"
                    stroke="#7e7e7e"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-sfs0yr-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-label="search"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  placeholder="Search…"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root css-2iaed4-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                id="demo-simple-select-label"
              >
                Short by:
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                data-testid="change-selection"
              >
                <div
                  aria-controls=":r0:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="demo-simple-select-label demo-simple-select"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="demo-simple-select"
                  role="combobox"
                  tabindex="0"
                >
                  -select-
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="default"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-10d36jd-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-18whfe4-MuiStack-root"
        >
          <div
            class="MuiStack-root css-156dzxc-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-ej953w-MuiTypography-root"
            >
              Customer Name
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-5khv0f-MuiTypography-root"
            >
              Company
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1fky39f-MuiTypography-root"
            >
              Phone Number
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-t6v2f4-MuiTypography-root"
            >
              Email
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-14ujwc1-MuiTypography-root"
            >
              Country
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-w9byz0-MuiTypography-root"
            >
              Status
            </h6>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-1bsgofg"
          >
            <div
              class="MuiStack-root css-1fbemys-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-caption css-bpkql1-MuiTypography-root"
              >
                John Doe
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption css-e8u01t-MuiTypography-root"
              >
                Company A
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption css-1mngl9u-MuiTypography-root"
              >
                1234567890
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption css-8zaaww-MuiTypography-root"
              >
                john@example.com
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption css-1nl03ot-MuiTypography-root"
              >
                USA
              </span>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-activate MuiButton-activatePrimary MuiButton-sizeMedium MuiButton-activateSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-activate MuiButton-activatePrimary MuiButton-sizeMedium MuiButton-activateSizeMedium MuiButton-colorPrimary css-1pyiguu-MuiButtonBase-root-MuiButton-root"
                data-testid="button-toggle"
                tabindex="0"
                type="button"
              >
                Active
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiBox-root css-79elbk"
              >
                <button
                  aria-haspopup="true"
                  aria-label="more"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="button-more-menu"
                  id="long-button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lrr7yx-MuiSvgIcon-root"
                    data-testid="MoreVertIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
            />
          </div>
          <div
            class="MuiStack-root css-bmu17z-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1jwn8bo-MuiTypography-root"
            >
              Showing data 
              1
               to
               
              1
               of
               
              1
               entries
            </h6>
            <nav
              aria-label="pagination navigation"
              class="MuiPagination-root MuiPagination-outlined css-1r9e4h5-MuiPagination-root"
              data-testid="change-pagination"
            >
              <ul
                class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
              >
                <li>
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateBeforeIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </button>
                </li>
                <li>
                  <button
                    aria-current="true"
                    aria-label="page 1"
                    class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-selected MuiPaginationItem-page css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                    tabindex="0"
                    type="button"
                  >
                    1
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </li>
                <li>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                    disabled=""
                    tabindex="-1"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                      data-testid="NavigateNextIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </button>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1dachdb"
    >
      <div
        class="MuiStack-root css-133lg72-MuiStack-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-15gon46-MuiTypography-root"
          >
            All Customers
          </h4>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1ge8cc0-MuiTypography-root"
          >
            Active Members
          </h6>
        </div>
        <div
          class="MuiStack-root css-qoybat-MuiStack-root"
        >
          <div
            class="MuiBox-root css-79elbk"
          >
            <div
              class="MuiStack-root css-1ibunwk-MuiStack-root"
            >
              <svg
                fill="none"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 19C15.4183 19 19 15.4183 19 11C19 6.58172 15.4183 3 11 3C6.58172 3 3 6.58172 3 11C3 15.4183 6.58172 19 11 19Z"
                  stroke="#7e7e7e"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
                <path
                  d="M21 21L16.65 16.65"
                  stroke="#7e7e7e"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-sfs0yr-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                placeholder="Search…"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root css-2iaed4-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-wxw8j3-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Short by:
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-cb5d54-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              data-testid="change-selection"
            >
              <div
                aria-controls=":r0:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-zbhh1g-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="combobox"
                tabindex="0"
              >
                -select-
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="default"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-10d36jd-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-18whfe4-MuiStack-root"
      >
        <div
          class="MuiStack-root css-156dzxc-MuiStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-ej953w-MuiTypography-root"
          >
            Customer Name
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-5khv0f-MuiTypography-root"
          >
            Company
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1fky39f-MuiTypography-root"
          >
            Phone Number
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-t6v2f4-MuiTypography-root"
          >
            Email
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-14ujwc1-MuiTypography-root"
          >
            Country
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-w9byz0-MuiTypography-root"
          >
            Status
          </h6>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1bsgofg"
        >
          <div
            class="MuiStack-root css-1fbemys-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-bpkql1-MuiTypography-root"
            >
              John Doe
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-e8u01t-MuiTypography-root"
            >
              Company A
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-1mngl9u-MuiTypography-root"
            >
              1234567890
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-8zaaww-MuiTypography-root"
            >
              john@example.com
            </span>
            <span
              class="MuiTypography-root MuiTypography-caption css-1nl03ot-MuiTypography-root"
            >
              USA
            </span>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-activate MuiButton-activatePrimary MuiButton-sizeMedium MuiButton-activateSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-activate MuiButton-activatePrimary MuiButton-sizeMedium MuiButton-activateSizeMedium MuiButton-colorPrimary css-1pyiguu-MuiButtonBase-root-MuiButton-root"
              data-testid="button-toggle"
              tabindex="0"
              type="button"
            >
              Active
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-79elbk"
            >
              <button
                aria-haspopup="true"
                aria-label="more"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                data-testid="button-more-menu"
                id="long-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1lrr7yx-MuiSvgIcon-root"
                  data-testid="MoreVertIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-yk10cl-MuiDivider-root"
          />
        </div>
        <div
          class="MuiStack-root css-bmu17z-MuiStack-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1jwn8bo-MuiTypography-root"
          >
            Showing data 
            1
             to
             
            1
             of
             
            1
             entries
          </h6>
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root MuiPagination-outlined css-1r9e4h5-MuiPagination-root"
            data-testid="change-pagination"
          >
            <ul
              class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateBeforeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-selected MuiPaginationItem-page css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                  tabindex="0"
                  type="button"
                >
                  1
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-ahvmcb-MuiButtonBase-root-MuiPaginationItem-root"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                    data-testid="NavigateNextIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
